hail/src/main/scala/is/hail/expr/types/physical/PArrayBackedContainer.scala

```diff
@@ -150,16 +150,15 @@ trait PArrayBackedContainer extends PContainer {
   def copyFromType(region: Region, srcPType: PType, srcAddress: Long, forceDeep: Boolean): Long =
     this.arrayRep.copyFromType(region, srcPType.asInstanceOf[PArrayBackedContainer].arrayRep, srcAddress, forceDeep)
 
-  def storeShallowAtOffset(dstAddress: Code[Long], valueAddress: Code[Long]): Code[Unit] =
-    this.arrayRep.storeShallowAtOffset(dstAddress, valueAddress)
-
-  def storeShallowAtOffset(dstAddress: Long, valueAddress: Long) {
-    this.arrayRep.storeShallowAtOffset(dstAddress, valueAddress)
-  }
-
   def nextElementAddress(currentOffset: Long) =
     arrayRep.nextElementAddress(currentOffset)
 
   def nextElementAddress(currentOffset: Code[Long]) =
     arrayRep.nextElementAddress(currentOffset)
+
+  def constructAtAddress(mb: MethodBuilder, addr: Code[Long], region: Code[Region], srcPType: PType, srcAddress: Code[Long], forceDeep: Boolean): Code[Unit] =
+    arrayRep.constructAtAddress(mb, addr, region, srcPType, srcAddress, forceDeep)
```

Comment:
In copyFromType on PArrayBackedContainer, I performed a cast on srcPType. That's now missing here, which could make the constructAtAddress fail (because srcPType's arrayRep isn't passed, which means arrayRep.constructAtAddress' type check will fail.

Reply:
hmm. Maybe the right thing is to enforce that all of these methods can ONLY be called on the fundamental type? Does that seem right?

Reply:
I think that is consistent with our semantics that fundamentalTypes are "constructible" types.

Reply:
yeah.